src/calendarevent.cpp

```diff
@@ -89,168 +99,228 @@ static Qt::TimeSpec toTimeSpec(const QDateTime &dt)
 
 Qt::TimeSpec CalendarEvent::startTimeSpec() const
 {
-    return toTimeSpec(mManager->getEvent(mUniqueId, mRecurrenceId).startTime);
+    return toTimeSpec(mData->startTime);
 }
 
 Qt::TimeSpec CalendarEvent::endTimeSpec() const
 {
-    return toTimeSpec(mManager->getEvent(mUniqueId, mRecurrenceId).endTime);
+    return toTimeSpec(mData->endTime);
 }
 
 QString CalendarEvent::startTimeZone() const
 {
-    return QString::fromLatin1(mManager->getEvent(mUniqueId, mRecurrenceId).startTime.timeZone().id());
+    return QString::fromLatin1(mData->startTime.timeZone().id());
 }
 
 QString CalendarEvent::endTimeZone() const
 {
-    return QString::fromLatin1(mManager->getEvent(mUniqueId, mRecurrenceId).endTime.timeZone().id());
+    return QString::fromLatin1(mData->endTime.timeZone().id());
 }
 
 bool CalendarEvent::allDay() const
 {
-    return mManager->getEvent(mUniqueId, mRecurrenceId).allDay;
+    return mData->allDay;
 }
 
 CalendarEvent::Recur CalendarEvent::recur() const
 {
-    return mManager->getEvent(mUniqueId, mRecurrenceId).recur;
+    return mData->recur;
 }
 
 QDateTime CalendarEvent::recurEndDate() const
 {
-    return QDateTime(mManager->getEvent(mUniqueId, mRecurrenceId).recurEndDate);
+    return QDateTime(mData->recurEndDate);
 }
 
 bool CalendarEvent::hasRecurEndDate() const
 {
-    return mManager->getEvent(mUniqueId, mRecurrenceId).recurEndDate.isValid();
+    return mData->recurEndDate.isValid();
 }
 
 CalendarEvent::Days CalendarEvent::recurWeeklyDays() const
 {
-    return mManager->getEvent(mUniqueId, mRecurrenceId).recurWeeklyDays;
+    return mData->recurWeeklyDays;
 }
 
 int CalendarEvent::reminder() const
 {
-    return mManager->getEvent(mUniqueId, mRecurrenceId).reminder;
+    return mData->reminder;
 }
 
 QDateTime CalendarEvent::reminderDateTime() const
 {
-    return mManager->getEvent(mUniqueId, mRecurrenceId).reminderDateTime;
+    return mData->reminderDateTime;
 }
 
 QString CalendarEvent::uniqueId() const
 {
-    return mUniqueId;
+    return mData->uniqueId;
 }
 
 QString CalendarEvent::color() const
 {
-    return mManager->getNotebookColor(mManager->getEvent(mUniqueId, mRecurrenceId).calendarUid);
+    return QString();
```

Comment:
Should this be in the base class if it doesn't return anything? Maybe at least could warrant some comment here.